docs/azure_jumpstart_ag/contoso_supermarket/freezer_monitor/_index.md

```diff
@@ -4,4 +4,264 @@ weight: 100
+
+In order to troubleshoot problems with the IoT data flow, it's important to understand the components involved so let's take a look at how the data flows from from a simulated freezer to the dashboards in Azure Data Explorer and Grafana.
+
+#### MQTT Simulator
+
+The first component, which generates the data for both dashboards is the MQTT Simulator. The simulator is a Python script that runs in each AKS Edge Essentials cluster. It generates simulated temperature and humidity data for two freezers in each environment and sends the data via the MQTT protocol to the MQTT Broker.
+
+To see data being produced by the MQTT Simulator
+
+From the Client VM:
+
+- Open __Visual Studio Code__
+- Click on the __Kubernetes__ icon in the Activity Bar on the left
+
+  ![Visual Studio Code showing the Kubernetes icon](./img/vscode_kubernetes_icon.png)
+- Right-click on the __Chicago__ cluster and select __Set as Current Cluster__
+
+  ![Visual Studio Code showing the Set as Current Cluster menu](./img/vscode_set_as_current_cluster.png)
+- Expand __Chicago__ > __Namespaces__, right-click on __sensor-monitor__ and select __Use Namespace__
+
+  ![Visual Studio Code showing the Use Namespace menu](./img/vscode_use_namespace.png)
+- Expand __Chicago__ > __Workloads__ > __Pods__ then right-click on the 'sensor-monitor-simulator-xxx' pod and select 'Logs'
+
+  ![Visual Studio Code showing the Logs menu](./img/vscode_simulator_logs_menu.png)
+- In the Logs view click __Run__ to see the logs
+
+  ![Visual Studio Code showing the Logs window](./img/vscode_simulator_logs_run.png)
+- This won't show you the values being produced, but it will show you that data is being published to the MQTT Broker
+
+  ![Visual Studio Code showing the Simulator logs](./img/vscode_simulator_logs.png)
+
+#### MQTT Broker
+
+The MQTT Broker is a container running Mosquitto in each AKS Edge Essentials cluster like the simulator. It receives the data from the simulator and sends it to the Azure IoT Hub. It also makes the data available for a third service, MQTT2Prometheus, which we'll discuss in a moment.
+
+To see data being received by the MQTT Broker
+
+Assuming you have completed the steps to view the MQTT Simulator logs above:
+
+- Expand __Chicago__ > __Workloads__ > __Pods__ then right-click on the 'sensor-monitor-simulator-xxx' pod and select 'Logs'
+
+  ![Visual Studio Code showing the Logs menu](./img/vscode_broker_logs_menu.png)
+- In the Logs view click __Run__ to see the logs
+
+  ![Visual Studio Code showing the Logs window](./img/vscode_broker_logs_run.png)
+- This won't show you the values being received or forwarded, but it will show you the connections from the 2 simulated freezer devices to the broker, as well as the connections from the broker to Azure IoT Hub for each freezer device. Finally, it shows the connection from __sensor-monitor-mqtt2prom__ which subscribes to the freezer data on the broker and makes it available to Prometheus, but more on that a bit later.
```

Comment:
```suggestion
- This won't show you the values being received or forwarded, but it will show you the connections from the 2 simulated freezer devices to the MQTT broker, as well as the connections from the MQTT broker to Azure IoT Hub for each freezer device. Finally, it shows the connection from __sensor-monitor-mqtt2prom__ which subscribes to the freezer data on the MQTT broker and makes it available to Prometheus and is explained in the sections below.
```